plenum/server/replica.py

```diff
@@ -1549,9 +1580,20 @@ def has_already_ordered(self, view_no, pp_seq_no):
     def isPpSeqNoBetweenWaterMarks(self, ppSeqNo: int):
         return self.h < ppSeqNo <= self.H
 
+
     def addToOrdered(self, viewNo: int, ppSeqNo: int):
         self.ordered.add((viewNo, ppSeqNo))
         self.last_ordered_3pc = (viewNo, ppSeqNo)
+        self.compact_ordered()
```

Comment:
Why call this after every ordering, calling it after view change complete it enough